File: cpp/Kode Program/SmartAppliance.cpp
#include "Appliance.cpp"
// Turunan dari Appliance.
// Tambah atribut buat elektronik pintar:
// jaringan (WiFi/BT), watt, sistem operasi, dan warna.
class SmartAppliance : public Appliance {
private:
    string network;
    int watt;
    string system;
    string color;

public:
    // Constructor lengkap
    SmartAppliance(int id, string name, string brand, int year,
                   string type, int warranty, double price, int stock,
                   string network, int watt, string system, string color)
        : Appliance(id, name, brand, year, type, warranty, price, stock) {
        this->network = network;
        this->watt = watt;
        this->system = system;
        this->color = color;
    }

    // Constructor default
    SmartAppliance() : Appliance() {
        this->network = "";
        this->watt = 0;
        this->system = "";
        this->color = "";
    }

    // Getter
    string getNetwork() { return network; }
    int getWatt() { return watt; }
    string getSystem() { return system; }
    string getColor() { return color; }

    // Setter
    void setNetwork(string n) { network = n; }
    void setWatt(int w) { watt = w; }
    void setSystem(string s) { system = s; }
    void setColor(string c) { color = c; }
};

File: java/Kode Program/SmartAppliance.java
public class SmartAppliance extends Appliance {
    private String network;
    private int watt;
    private String system;
    private String color;

    // Constructor lengkap
    public SmartAppliance(int id, String name, String brand, int year,
                          String type, int warranty, double price, int stock,
                          String network, int watt, String system, String color) {
        // super() → manggil constructor Appliance
        super(id, name, brand, year, type, warranty, price, stock);
        this.network = network;
        this.watt = watt;
        this.system = system;
        this.color = color;
    }

    // Constructor default
    public SmartAppliance() {
        super(); // manggil constructor default Appliance
        this.network = "";
        this.watt = 0;
        this.system = "";
        this.color = "";
    }

    // Getter
    public String getNetwork() { return network; }
    public int getWatt() { return watt; }
    public String getSystem() { return system; }
    public String getColor() { return color; }

    // Setter
    public void setNetwork(String n) { this.network = n; }
    public void setWatt(int w) { this.watt = w; }
    public void setSystem(String s) { this.system = s; }
    public void setColor(String c) { this.color = c; }
}
